contracts/Governance/Governor.sol

```diff
@@ -288,8 +288,8 @@ contract Governor {
     function cancel(uint256 proposalId) public {
         ProposalState _state = state(proposalId);
         require(
-            _state != ProposalState.Executed,
-            "Governor::cancel: cannot cancel executed proposal"
+            _state == ProposalState.Queued,
+            "Governor::cancel: cannot cancel proposal that isn't queued"
         );
```

Comment:
This change comes from the trail of bits audit.

Reply:
It doesn't make sense to me, why shouldn't the guardian be able to shoot down proposals in other states?

Reply:
No, I actually don't see it. If the proposer loses power to push forward changes, why do we need to wait for their proposal to reach this late state to acknowledge it?

Reply:
So the thing is, if you check the Timelock cancelTransaction code and queueTransaction, if we call cancel before it's queued, it's still possible to queue, and then that will just "undo" the cancel, in the timelock.

Reply:
I agree that canceling something not queued on the timelock is a noop. But cancelling also sets the state of the proposal on the governor side. For example cancelling a proposal, which is active should prevent it from being able to be voted on and proceeded.